Rubberduck.Core/UI/Command/ExportAllCommand.cs

```diff
@@ -32,17 +31,32 @@ protected override bool EvaluateCanExecute(object parameter)
                 return false;
             }
 
-            var projectNode = parameter as CodeExplorerProjectViewModel;
-
-            var project = parameter as IVBProject;
-
-            return Evaluate(projectNode?.Declaration.Project ?? project ?? _vbe.ActiveVBProject);
+            switch (parameter)
+            {
+                case CodeExplorerProjectViewModel projectNode:
+                    return Evaluate(projectNode.Declaration.Project);
+                case IVBProject project:
+                    return Evaluate(project);
+            }
 
+            using (var activeProject = _vbe.ActiveVBProject)
+            {
+                return Evaluate(activeProject);
+            }
         }
 
         private bool Evaluate(IVBProject project)
         {
-            return project != null && !project.IsWrappingNullReference && project.VBComponents.Count > 0;
+            if (project == null || project.IsWrappingNullReference)
```

Comment:
Not really in scope for the PR (commenting for my own reference), but this wants to be a generic extension someplace due to how many places we make this check. I'm thinking something along the lines of:

```
        public static bool IsLiveWrapper<T>(this SafeComWrapper<T> wrapper) where T : class
        {
            return wrapper != null && !wrapper.IsWrappingNullReference;
        }
```

Reply:
Doesn't the `SafeComWrapper` override `==` anyway to mean this?

Reply:
The `==` override doesn't check whether it's wrapping a null reference.

Reply:
Actually, it does but in an indirect way that makes the `==` asymmetrical. When `a` in `a == b` is not `null'', then the override returns `a.Equals(b)`. The implementation in `VBProject` uses `IsEqualIfNull` inherited from `SafeComWrapper`. This in turn returns true if `a` is wrapping a null reference.

I have to say that I really do not like that the operator is asymmetric in this way and that the derived classes can choose to honor this intended behaviour or not. (If it was not intended, why would there be the protected method `IsEqualIfNull`?)